docs/patterns.rst

```diff
@@ -114,15 +118,119 @@ a ``categorical`` slot.
 
 Rasa automatically represents (featurises) this as a one-hot encoding of the values: ``(1,0,0)``, ``(0,1,0)``, or ``(0,0,1)``.
 
+.. _slot_features:
+
+**Slot features**
+
+When Rasa Core runs trains a dialogue model using your stories the presence of a ``Slot`` entry will be used to help 
+determine the next action that should be taken. This works best with ``CategoricalSlot`` slot types. 
+
+A ``TextSlot`` can have any value, but it only has a single feature. It can be set, in which case the feature 
+has a value ``(1)``, or if it is not set it will have a value ``(0)``. 
+
+A ``CategoricalSlot`` has a number of values each of which is a feature. Taking the example below, when the 
+``restaurant_availability`` slot is set Rasa Core will be able to determine whether or not the restaurant in question is 
+available and choose radically different actions to perform based on the value.
+
+.. code-block:: yaml
+
+    restaurant_availability:
+        type: categorical
+        values:
+        - unknown
+        - booked-out
+        - waiting-list
+        - available
+
+A ``Slot`` will be set by Rasa Core if its name and the name of the entity detected by the NLU module match. However, when we wish to alter the flow of a conversation we would explicitly set the value of the slot from a custom ``Action`` and 
```

Comment:
this isn't quite right. whether the slot is set automatically or in a custom action, it has the same ability to impact the conversation flow. So long as you have relevant training data, rasa core will pick up on these patterns